lumicks/pylake/channel.py

```diff
@@ -239,6 +239,65 @@ def calibration(self) -> list:
         else:
             return []
 
+    def recalibrate_force(self, calibration):
+        """Recalibrate the data in this slice using a new force calibration item
+
+        Parameters
+        ----------
+        calibration : ForceCalibrationItem
+            The calibration to apply to the data
+
+        Examples
+        --------
+        ::
+
+            import lumicks.pylake as lk
+
+            f = lk.File("passive_calibration.h5")
+            calibration = f.force1x.calibration[1]  # Grab a calibration item for force 1x
+
+            # Slice the data corresponding to the calibration we want to reproduce.
+            calib_slice = f.force1x[calibration]
+
+            # De-calibrate to volts using the calibration that was active before this slice.
+            previous_calibration = calib_slice.calibration[0]
+            calib_slice = calib_slice / previous_calibration.force_sensitivity
+
+            calibration_params = previous_calibration.calibration_params()
+            new_calibration = lk.calibrate_force(calib_slice.data, **calibration_params)
+            new_calibration.plot()
+
+            # Make a new calibration, but change the amount of blocking
+            less_blocking_params = calibration_params | {"num_points_per_block": 200}
+            less_blocking = lk.calibrate_force(calib_slice.data, **less_blocking_params)
+            less_blocking.plot()
+
+            # Recalibrate the force channels
+            recalibrated_force1x = f.force1x.recalibrate_force(less_blocking)
+        """
+        from lumicks.pylake.calibration import ForceCalibrationList
+
+        if not self.calibration:
+            raise RuntimeError(
+                "Slice does not contain any calibration items. Is this an unprocessed force "
+                "channel?"
+            )
+
+        # Grab sensitivities. If for whatever reason, we don't know part of the calibration, we
+        # can't reasonably recalibrate, so we return NaNs for those time points.
+        timestamps = self.timestamps
+        force_sensitivities = np.full(len(timestamps), np.nan)
+        for c in self.calibration:
+            force_sensitivities[timestamps >= c.applied_at] = c.force_sensitivity
```

Comment:
is it that expensive here to look up `timestamps` twice-ish that it should be a separate variable? I can't imagine a slice with like 50 calibrations where this loop would run signifcantly long. If not, I think I prefer to just use `self.timestamps` at each location

Reply:
Not expensive. I'll swap it out.